### ISSUE TITLE
Doesn't work with git-cmd

### ISSUE THREAD

**Issue by kt215** [Tyrrrz/CliWrap#21]
### Thanks for creating CliWrap :-)

When I run `git-cmd.exe` it **successfully completes**, but `ExecutionOutput.StandardError` is still filled up with text and the `ExecutionOutput.HasError` is `true`. Is this a bug in CliWrap or is there something wrong with `git-cmd.exe`?

**How to reproduce:**

1. On Windows 10, download and install `Git-2.16.2-64-bit.exe` from https://git-scm.com/download/win
2. Make sure `C:\temp` folder exists.
2. Run this code:

```cs
using (var cli = new Cli(@"C:\Program Files\Git\git-cmd.exe", new CliSettings { WorkingDirectory = @"C:\temp" })) {
    var output = await cli.ExecuteAsync("git clone -b \"1.8.2\" --single-branch --depth 1 https://github.com/Tyrrrz/CliWrap.git");
    var errortext = output.StandardError;
}
```

BTW, the StandardError contains this text:

```
Cloning into 'CliWrap'...
remote: Counting objects: 38, done.
remote: Compressing objects: 100% (34/34), done.
remote: Total 38 (delta 1), reused 18 (delta 0), pack-reused 0
Unpacking objects: 100% (38/38), done.
Note: checking out 'a4f337c2e4549ae5866656f8249cb21d51794f17'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>
```

**Comment by Tyrrrz** [comment]
Hey, thank you for the report and kind words.

Judging from your description, it looks like `git.exe` is sending all output to StdErr for some reason. I'll try to reproduce using the steps you've given a bit later today.

**Comment by Tyrrrz** [comment]
It appears `git clone` always write to stderr, for some reason -- https://stackoverflow.com/questions/34820975/git-clone-redirect-stderr-to-stdout-but-keep-errors-being-written-to-stderr.

**Comment by kt215** [comment]
You are correct. After further testing, if the git label is wrong, StandardError starts with `warning:` then followed by `fatal:`... So checking `ExitCode` makes the code more robust:

```cs
using (var cli = new Cli(@"C:\Program Files\Git\bin\git.exe", new CliSettings { WorkingDirectory = @"C:\temp" })) {
    var output = await cli.ExecuteAsync("clone -b \"1.8.2\" --single-branch --depth 1 https://github.com/Tyrrrz/CliWrap.git");
    if (output.ExitCode != 0) {
        output.ThrowIfError();
    }
}
```
With this in mind, perhaps the `HasError` property in `ExecutionOutput.cs` should be implemented like this:

```cs
public bool HasError => ExitCode != 0 || !string.IsNullOrEmpty(StandardError);
```
or

```cs
public bool HasError => ExitCode != 0 || (ReportErrorsWithNormalExitCode && !string.IsNullOrEmpty(StandardError));
```
where `ReportErrorsWithNormalExitCode` is a new option to handle CLI tools like `git.exe`.

What do you think?

**Comment by Tyrrrz** [comment]
I don't really want to make `HasError` sophisticated. Exit code 0 is just the default, it doesn't really guarantee success, the same way (as you noticed) writing to stderr doesn't guarantee failure. There are standards and some CLIs just throw them out of the window. So I think it's best to have `HasError` tied to `StandardError` and, if need be, the users can implement custom logic, like you did.

**Comment by Tyrrrz** [comment]
On the other hand, having multiple error detection strategies that you can select from in `CliSettings` could be nice to have. Please submit a PR if you want to implement something like that.

**Comment by kt215** [comment]
Ok, I'll consider that. Thanks.
